lm_survey/samplers/base_sampler.py

```diff
@@ -40,11 +46,18 @@ def get_best_next_token(self, prompt: str, **kwargs) -> str:
         Return:
             str a generated sequence
         """
-        logprobs = self.send_prompt(prompt=prompt, n_probs=1, **kwargs)
-        return list(logprobs.keys())[0]
+        if not asyncio.iscoroutinefunction(self.send_prompt):
```

Comment:
TODO: This won't work; The way AutoSampler is set up makes this a little bit tricky. Getting this to work on AutoSampler would force us to use __getattr__ to get the method, which is a problem because then we get an error about not providing implementations for BaseSampler's abstract methods. Which leaves 3 options:

1. Make AutoSampler stop subclassing BaseSampler.
2. This, which makes AutoSampler stop subclassing BaseSampler only at runtime:
    ```python
    from typing import TYPE_CHECKING
    BaseClass = BaseSampler if TYPE_CHECKING else object
    ```
3. Come up with an actually fleshed out async implementation.

That said, we don't use this anywhere in the code right now…